File: tarea-2/Program/Items/Comida.java
package Program.Items;

import Program.Mascota;

public final class Comida extends Item{
    String tipo = "COMIDA";

    public Comida(int id, String nombre, int cantidad){
        super(id, nombre, cantidad);

    }

    // usar comida => energía y vida de la mascota aumentan +20, luego consume el item (eliminado de inventario)
    @Override
    public void usarItem(Mascota mascota){
        double newSalud = mascota.getSalud().get() + 20;
        double newEnergia = mascota.getEnergia().get() + 20;

        if (newSalud > Mascota.MAX_STATS) {newSalud = Mascota.MAX_STATS;}
        if (newEnergia > Mascota.MAX_STATS) {newEnergia = Mascota.MAX_STATS;}

        mascota.setSalud(newSalud);
        mascota.setEnergia(newEnergia);

        cantidad--;
    }

    @Override
    public String getTipo() {return tipo;}

    @Override
    public Item clone() {
        Item clone = new Comida(id, nombre, cantidad);
        return clone;
    }
}

File: tarea-3/etapa-2/Comida.cpp
#include <iostream>
#include <string>
#include "../headers/Comida.h"
#include "../headers/Mascota.h"

Comida::Comida(int i, std::string n, int c) : Item(i, "Comida", n, c) {}
void Comida::usarItem(Mascota &mascota) {
	std::cout << "Usando " << tipo << " " << nombre << " en la Mascota" << std::endl;

	int newSalud = mascota.getSalud() + 20;
	int newEnergia = mascota.getEnergia() + 20;

	if (newSalud > MAX_STATS) {newSalud = MAX_STATS;}
	if (newEnergia > MAX_STATS) {newEnergia = MAX_STATS;}

	mascota.setSalud(newSalud);
	mascota.setEnergia(newEnergia);
	cantidad--;
	mascota.mostrarMascota();
} 

